### PR TITLE
simplify connectivity check

### PR DESCRIPTION
**Pull request by olamy**
- for anonymous check we may have not the right to getAttributes so we must simply ensure we can connect
- timeout should be configurable, simplify some code as well

### CODE REVIEW

File: src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java [hunk 1]
```diff
@@ -573,8 +599,9 @@ public ApplicationContext createApplicationContext(LDAPSecurityRealm realm) {
         }
         contextSource.setReferral("follow");
         Map<String, Object> vars = new HashMap<>();
-        vars.put("com.sun.jndi.ldap.connect.timeout", "30000"); // timeout if no connection after 30 seconds
-        vars.put("com.sun.jndi.ldap.read.timeout", "60000"); // timeout if no response after 60 seconds
+        vars.put("com.sun.jndi.ldap.connect.pool", "true");
```
Comment (jtnord) [inline src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java:602]: Are we sure we want this, especially given it is not configurable?

Comment (jtnord) [reply src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java:602]: https://docs.spring.io/spring-ldap/docs/1.3.2.RELEASE/reference/html/pooling.html. Implies this poolingbis bad and there is no validation of the context.

File: src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java [hunk 2]
```diff
@@ -403,19 +410,25 @@ public FormValidation doCheckServer(@QueryParameter String value, @QueryParamete
...
                 }
+                // normal
                 props.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
                 props.put(Context.PROVIDER_URL, LDAPSecurityRealm.toProviderUrl(server,rootDN));
 
-                DirContext ctx = new InitialDirContext(props);
-                ctx.getAttributes("");
+                props.put("java.naming.referral", "follow");
+                props.put("com.sun.jndi.ldap.connect.timeout", Integer.toString(CONNECT_TIMEOUT));
+                props.put("com.sun.jndi.ldap.connect.pool", "true");
```
Comment (jtnord) [inline src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java:428]: Pooling a one of check makes no sense

File: src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java (hunk 1 above)
Comment (olamy) [reply src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java:602]: the idea was to have the exact same code/configuration for this test and how connection are created when used for authz see https://github.com/jenkinsci/ldap-plugin/blob/992524769cc9616d69b035b247d21badd9baafc1/src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java#L602
it was totally different which is not better

File: src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java (hunk 2 above)
Comment (olamy) [reply src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java:428]: answer here https://github.com/jenkinsci/ldap-plugin/pull/102#discussion_r705056385

File: src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java (hunk 1 above)
Comment (jtnord) [reply src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java:602]: AHH, that may explain some random hangs. Of. The plugin then that are fixed by restarting.
I suggest pooling the s disabled everywhere, or isboptbin

Comment (olamy) [reply src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java:602]: well sounds good. disabled per default but a configurable option?

Comment (olamy) [reply src/main/java/jenkins/security/plugins/ldap/LDAPConfiguration.java:602]: well there is still an option to add env props

**Pull request merged**